### PR TITLE
Added example of a simple mixin

### LINKED ISSUE #136: Mixins: Simple extend/override
A runnable example show-casing how one can create simple mixins.

Path: /examples/mixins/simple-extend-override

### PR DESCRIPTION
**Pull request by peol**
In addition, a few internal properties was renamed to their proper external names

Fixes #136

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -426,7 +438,7 @@ doc.getObject('object-id').then((api) => {
 
 [Back to top](#api-documentation)
 
-### `api.customType`
+### `api.genericType`
 
 This property contains the custom type set on this API (if any).
```
Comment (marcusoffesson) [inline docs/api.md]: Possible clarification:

`Despite the name, this property contains the custom type ...`

File: examples/mixins/simple/simple.js [hunk 2]
```diff
@@ -52,18 +32,21 @@ const docMixin = {
   extend: {
     /**
     * Simple tweeting function.
+    * @returns {Promise<Object|null>} A promise that when resolved means this mixin method
```
Comment (marcusoffesson) [inline examples/mixins/simple/simple.js]: `Promise<null>`?

File: examples/mixins/simple/simple.js [hunk 3]
```diff
@@ -72,30 +55,30 @@ const docMixin = {
...
     /**
-    * Overriding the getObject function.
+    * Overriding the createObject function.
     * @param {Function} base - This is the original function that is being overridden.
-    *                          Can be used in the override.
     * @param {*} params - The parameter list. When parameters are passed by name, enigma.js
     *                     will add default values for parameters not supplied by the caller.
-    * @returns {Promise<Object|null>} A promise that when resolved contains the object asked
-    *                                 for or null if object doesn't exist.
+    * @returns {Promise<Object|null>} A promise that when resolved contains the newly created
```
Comment (marcusoffesson) [inline examples/mixins/simple/simple.js]: `Promise<Object>`?

File: examples/mixins/simple/simple.js (hunk 2 above)
Comment (peol) [inline examples/mixins/simple/simple.js]: It tries to clarify that you don't need to resolve the promise with a value, I'll change these to simply `Promise` to be consistent with other examples.

**Pull request merged**